### PR TITLE
feat(ui5-form): update to latest accessibility spec

### PR DESCRIPTION
**Pull request by ilhan007**
The following accessibility changes are applied to the Form web component to full-fill the latest spec.

**Form**

- form with groups has role="region"
- form without groups have role="form"
- form are referenced with the form header text (via aria-labelledby)

**FormGroup**

- form groups are referenced with their form group header texts (via aria-labelledby)
- form groups have role="form"
- form groups create a fast navigation group - to jump from one group to another with F6

Fixes: https://github.com/SAP/ui5-webcomponents/issues/9952

### CODE REVIEW

**Review by nnaydenow — COMMENTED** [review]
What about form with only 1 group? Should it has role "region" or "form"?

File: packages/main/src/Form.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ type GroupItemsInfo = {
 	groupItem: IFormItem,
 	classes: string,
 	items: Array<ItemsInfo>,
+	ariaLabelledBy: string | undefined
```
Comment (nnaydenow) [inline packages/main/src/Form.ts:47]: probably this type conflicts too with the definition of aria-labelledby native attribute.

File: packages/main/src/Form.ts [hunk 2]
```diff
@@ -365,10 +366,15 @@ class Form extends UI5Element {
 		return this.hasCustomHeader ? undefined : `${this._id}-header-text`;
 	}
 
+	get ariaRole(): string | undefined {
```
Comment (nnaydenow) [inline packages/main/src/Form.ts:369]: rename to effectiveAccessibleRole in order to not conflict with the native ariaRole typescript definition.

**Review by nnaydenow — CHANGES_REQUESTED** [review]
LGTM as discussed offline.

File: packages/main/src/Form.ts [hunk 3]
```diff
@@ -301,6 +309,17 @@ class Form extends UI5Element {
 		});
 	}
 
+	setFastNavGroups() {
+		if (!this.hasGroupItems) {
+			this.setAttribute("data-sap-ui-fastnavgroup", "true");
+			return;
+		}
+
+		this.items.forEach((item: IFormItem) => {
```
Comment (nnaydenow) [inline packages/main/src/Form.ts:312]: this could be moved to customElement decorator of ui5-form-group

File: packages/main/src/Form.ts [hunk 4]
```diff
@@ -301,6 +309,17 @@ class Form extends UI5Element {
 		});
 	}
 
+	setFastNavGroups() {
+		if (!this.hasGroupItems) {
+			this.setAttribute("data-sap-ui-fastnavgroup", "true");
```
Comment (nnaydenow) [inline packages/main/src/Form.ts:308]: probably we need to remove attribute when the slot is changed and `hasGroupItem` returns true

File: packages/main/src/Form.ts (hunk 1 above)
Comment (ilhan007) [reply packages/main/src/Form.ts]: done

File: packages/main/src/Form.ts (hunk 2 above)
Comment (ilhan007) [reply packages/main/src/Form.ts]: done

**Review by nnaydenow — CHANGES_REQUESTED** [review]

File: packages/main/src/Form.ts (hunk 3 above)
Comment (ilhan007) [reply packages/main/src/Form.ts]: fixed

**Review by nnaydenow — APPROVED** [review]

**Pull request merged**